react/modules/tasks/forms.md

```diff
@@ -18,29 +23,63 @@
    - Approach with **React Hook Form** should implement live validation
-5. After submitting the form
-   On successful form submission redirect user to the main route with all the previously entered data. Make an indication for a newly entered data on the main route (e.g. show border in a different color for a few seconds, or a different background color)
+
+5. **After submitting the form**
+   On successful form submission close the modal and display the newly entered data on the main page. Make an indication for a newly entered data on the main route (e.g. show border in a different color for a few seconds, or a different background color)
+
+6. **Testing Requirements / Unit Testing**
+
+   - **Form Components Testing**: Test both uncontrolled and React Hook Form implementations
+
+     - Test form rendering with all required fields
+     - Test field validation (name, age, email, passwords, etc.)
+     - Test password strength calculation
+     - Test form submission with valid/invalid data
+     - Test error message display and clearing
+
+   - **Modal Components Testing**:
+
+     - Test modal opening/closing functionality
+     - Test accessibility features (focus management, ESC key)
+     - Test click outside to close behavior
+     - Test portal rendering
+
+   - **Redux Store Testing**:
+
+     - Test actions and action creators
+     - Test reducers with different action types
+     - Test selectors
+     - Test store state updates after form submissions
+
+   - **Utility Functions Testing**:
+     - Test password strength validation
+     - Test image to base64 conversion
+     - Test form validation helpers
+     - Test country autocomplete filtering
 
 ## Technical Requirements
 
 1. Create a separate branch for this task. Branch name: "forms". For this task you will need to create a new application.
 2. Follow the requirements for the project setup listed [here](./project-setup.md)
-3. Add **React Hook Form**, **Yup** or **Zod**, **Redux Toolkit** and **React Router** to the application.
+3. Add **React Hook Form**, **Yup** or **Zod**, **Redux Toolkit**, **Jest** or **Vitest** with React Testing Library to the application.
 
 ## Points
 
 ### Student can get 100 points:
 
-- 3 routes (main and 2 routes for forms), Redux is set up and used to collect data from both forms, redirect to main route after submitting the form - **15 points**
-- Validation works for both forms according to the requirements (error messages, blocking submit button), **Yup** or **Zod** is used for validation (10 points if works only for one form) - **20 points**
-- Name, age, email, gender picker, accept T&C are implemented for both forms and collect data (if something doesn't work, score can be less) - **20 points**
-- Input for image is implemented for both forms, image is saved as base64 and dispaly on the main route after redirect - **15 points**
+- 2 modals using React Portals (one for each form approach) - **15 points**
```

Comment:
I guess there could be just one modal, which displays children, so both forms could reuse the same modal.